corehq/apps/consumer_user/tasks.py

```diff
@@ -0,0 +1,131 @@
+from datetime import datetime, timedelta
+from collections import defaultdict
+
+from django.template.loader import render_to_string
+from django.utils.translation import ugettext_lazy as _
+
+from celery.schedules import crontab
+from celery.task import periodic_task, task
+
+from corehq.apps.consumer_user.models import (
+    ConsumerUserCaseRelationship,
+    ConsumerUserInvitation,
+)
+from corehq.apps.hqcase.utils import update_case, bulk_update_cases
+from corehq.apps.hqwebapp.tasks import send_html_email_async
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.util.view_utils import absolute_reverse
+
+from .const import (
+    CONSUMER_INVITATION_ACCEPTED,
+    CONSUMER_INVITATION_ERROR,
+    CONSUMER_INVITATION_SENT,
+    CONSUMER_INVITATION_STATUS,
+)
+
+
+@task
+def handle_consumer_user_invitations(domain, invitation_case_id, demographic_case_id):
+    invitation_case = CaseAccessors(domain).get_case(invitation_case_id)
+    email = invitation_case.get_case_property('email')
+    status = invitation_case.get_case_property(CONSUMER_INVITATION_STATUS)
+    keep_open_status = [CONSUMER_INVITATION_SENT, CONSUMER_INVITATION_ACCEPTED]
+
+    if not invitation_case.closed and ConsumerUserInvitation.objects.filter(
+        demographic_case_id=demographic_case_id, active=True, accepted=False
+    ).exclude(case_id=invitation_case_id).exists():
+        # If there is currently an open, unaccepted invitation for the same demographic case, do nothing
+        update_case(
+            domain=domain,
+            case_id=invitation_case_id,
+            case_properties={
+                CONSUMER_INVITATION_STATUS: CONSUMER_INVITATION_ERROR,
+                CONSUMER_INVITATION_ERROR: "There is already an open invitation for this case",
+            },
+            device_id=__name__ + '.handle_consumer_user_invitations',
```

Comment:
is it by design that the device ID would change if this code ever moved? guessing this is following an existing convention?